File: Array-II/Q1.Set Mismatch/findErrorNums.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function (nums) {
    const len = nums.length;
    let double = 0
    let missed = 0

    nums.sort((a, b) => a - b)

    //get the doublicate number
    for (let i = 0; i < len - 1; i++) {
        if (nums[i] == nums[i + 1]) {
            double = nums[i]
            break
        }
    }

    //get the missed number
    for (let i = 1; i <= len; i++) {
        if (!nums.includes(i)) {
            missed = i
            break
        }
    }

    return [double, missed];
};

//SECOND SOLUTION USING HASH-TABLE
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findErrorNums = function (nums) {
    const len = nums.length;
    let double = 0;
    let missed = 0;
    let hash = {}

    for (let i = 1; i <= len; i++) {
        hash[i] = 0
    }

    for (let num of nums) {
        hash[num] += 1
    }

    for (let key in hash) {
        if (hash[key] > 1) {//2
            double = parseInt(key)
        }
        if (hash[key] < 1) {//0
            missed = parseInt(key)
        }
    }

    return [double, missed]
};

File: Array-II/Q1.Set Mismatch/findErrorNums.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int sz = nums.size();
        sort(nums.begin(), nums.end());
        vector<int> ans;

        for(int i=0; i<sz-1; i++){
            if(nums[i] == nums[i+1]){
                ans.push_back(nums[i]);
                break;
            }
        }
        for(int i=1 ; i<=sz; i++){
            auto it = find(nums.begin(), nums.end() , i);
            if(it == nums.end()){
                ans.push_back(i);
                break;
            }
        }

        
        return ans;
    }
};


/*
[3,2,2] => SORT
[2,2,3]
*/

/*
[3,2,3,4,6,5] => SORT
[2,3,3,4,5,6]
*/